src/shared/tpm2-util.c

```diff
@@ -5431,6 +5431,24 @@ char *tpm2_pcr_mask_to_string(uint32_t mask) {
         return TAKE_PTR(s);
 }
 
+const uint16_t tpm2_hash_algorithms[] = {
+        TPM2_ALG_SHA1,
+        TPM2_ALG_SHA256,
+        TPM2_ALG_SHA384,
+        TPM2_ALG_SHA512,
+        0,
+};
+
+assert_cc(ELEMENTSOF(tpm2_hash_algorithms) == TPM2_N_HASH_ALGORITHMS + 1);
+
+size_t tpm2_hash_algorithm_index(uint16_t algorithm) {
```

Comment:
this seems strange, and if it's only used for referencing the correct `Tpm2PCRPredictionResult->hash[]`, maybe it would be better to just have something like `TPM2B_DIGEST *tpm2_pcr_prediction_result_get_hash(const Tpm2PCRPredictionResult *result, uint16_t algorithm)` which would return a pointer to the correct hash (or NULL if `algorithm` is invalid)? That would at least keep the alg<->index mapping internal to the `Tpm2PCRPredictionResult` object (i.e. all users would always reference the result's hashes only by algorithm, instead of by index)

Reply:
updated as suggested.